includes/embeds/class-amp-playlist-embed-handler.php

```diff
@@ -0,0 +1,307 @@
+<?php
+/**
+ * Class AMP_Playlist_Embed_Handler
+ *
+ * @package AMP
+ * @since 0.7
+ */
+
+/**
+ * Class AMP_Playlist_Embed_Handler
+ *
+ * Creates AMP-compatible markup for the WordPress 'playlist' shortcode.
+ *
+ * @package AMP
+ */
+class AMP_Playlist_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * The tag of the shortcode.
+	 *
+	 * @var string.
+	 */
+	const SHORTCODE = 'playlist';
+
+	/**
+	 * The max width of the audio thumbnail image.
+	 *
+	 * This corresponds to the max-width in wp-mediaelement.css:
+	 * .wp-playlist .wp-playlist-current-item img
+	 *
+	 * @var string.
+	 */
+	const THUMB_MAX_WIDTH = 60;
+
+	/**
+	 * The height of the carousel.
+	 *
+	 * @var string.
+	 */
+	const CAROUSEL_HEIGHT = 160;
+
+	/**
+	 * The pattern to get the playlist data.
+	 *
+	 * @var string.
+	 */
+	const PLAYLIST_REGEX = '/(?s)\<script [^>]* class="wp-playlist-script"\>[^<]*?(.*).*?\<\/script\>/';
```

Comment:
I don't think this regex is quite right. This should be more robust:

```php
const PLAYLIST_REGEX = ':<script type="application/json" class="wp-playlist-script">(.+?)</script>:s';
```

Reply:
Thanks, [this commit](https://github.com/Automattic/amp-wp/pull/954/commits/f13d2ad1f8e7d1a2c8495b865a6a0371ad96709d) applies your regex.